### DISCUSSION TITLE
How to save ssh connections?

### DISCUSSION THREAD

**Discussion by im-mansoor** [Eugeny/tabby#5015] [Q&A]
Hello there

I need help to save my ssh connections as a group
i have several servers and need to save all servers in groups so i can connect to them whenever needed any of them.
also i need to arrange multiple servers in same page (split each one beside another. but couldn't find a way for any of them.
how can i do this?

Thanks in advance for your kind help

**Comment by Eugeny** [comment]
> i have several servers and need to save all servers in groups so i can connect to them whenever needed any of them.

Settings -> Connections & Profiles - you can create and assign profiles into groups there

> also i need to arrange multiple servers in same page (split each one beside another. but couldn't find a way for any of them.
how can i do this?

Connect to both servers, then drag the other server's tab onto the side of the active tab. This will combine them into a single split tab. (alternatively you can also use the "Split to left/right/top/bottom" hotkeys and use the "switch profile" context menu item.

To recall it later, right-click the now-split tab and select "save layout as profile".

**Reply by im-mansoor** [reply]
> 
> Settings -> Connections & Profiles - you can create and assign profiles into groups there
> 

I've created groups there but unable to add ssh servers there!


> Connect to both servers, then drag the other server's tab onto the side of the active tab. This will combine them into a single split tab. (alternatively you can also use the "Split to left/right/top/bottom" hotkeys and use the "switch profile" context menu item.
> 
> To recall it later, right-click the now-split tab and select "save layout as profile".

Thanks a lot. first part (combining currently open tabs) worked with no problem but after saving layout as profile, when i wanted to restore them, it just opens an empty terminal for me.

what I've did:

- Connected to 3 servers at same time
- Combined them by dragging tabs into single tab
- Right clicked on tab name & clicked on "Save layout as profile"

after reopening tabby, went to Settings ~> Profiles & connections ~> clicked on play icon (triangle) in front of layout saved in previous step.

**Reply by Eugeny** [reply]
> I've created groups there but unable to add ssh servers there!
You're setting the group by opening a profile and selecting it in the Group field, right?

Which Tabby version are you on?

**Reply by Eugeny** [reply]
> after reopening tabby, went to Settings ~> Profiles & connections ~> clicked on play icon (triangle) in front of layout saved in previous step.

Please capture DevTools console logs at the time of the issue: https://github.com/Eugeny/terminus/wiki/Debug-logs

**Reply by im-mansoor** [reply]
> 

well, i've found sometyhing there but not sure how this can cause the problem & how should i solve it.

Error in console:
`GET file:///opt/Tabby/resources/app.asar/dist/undefined net::ERR_FILE_NOT_FOUND`

Content of warning: https://paste.centos.org/view/782f7579

Tabby version: 1.0.164

**Comment by im-mansoor** [comment]
Seems it was because of a plugin called "Workspace Manager"
after disabling this, now starting profile is reopenning terminals of previous stem without any problem. so that problem is solved now. thanks to you.

But the main question is still remain, how can i save connections to connect to them later.
take a look at this link: https://mobaxterm.mobatek.net/

this is what i need exactly to make a category and save each connection below that category. so whenever i want to connect, i should simply click on the server i want.

**Reply by Eugeny** [reply]
Could you explain how exactly you can't save connections?
